### PR TITLE
[!!!]Replace updater with a new system for placing reservations for not paid orders from Allegro

### PR DESCRIPTION
**Pull request by mstrzyzewski**
-add logic for placing and compensating reservations
-add a new table in the database for keeping information about Allegro reservations
-create new admin grid for listing Allegro reservations
-delete logic for updating orders

### CODE REVIEW

File: Controller/Adminhtml/Reservations/Index.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+use Magento\Backend\App\Action;
+use Magento\Framework\App\ResponseInterface;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\View\Result\Page;
+
+/**
+ * Index controller class
+ */
+abstract class Index extends Action
```
Comment (cieslix) [inline Controller/Adminhtml/Reservations/Index.php]: why `abstract class` Index ?

File: Model/OrderImporter/AllegroReservation.php [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    {
+        try {
+            if (!$this->configuration->areReservationsEnabled()) {
+                return;
+            }
+            $this->placeReservationsForSalesEvent->execute(
+                $this->getItemsToSell($checkoutForm, false),
+                $this->getSalesChannel(),
+                $this->getSalesEvent(self::ALLEGRO_EVENT_RESERVATION_PLACED, $checkoutForm->getId()));
+        } catch (\Exception $e) {
```
Comment (cieslix) [inline Model/OrderImporter/AllegroReservation.php]: why catch and throw same exception?

File: Model/OrderImporter/AllegroReservation.php [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+        try {
+            if (!$this->configuration->areReservationsEnabled()) {
+                return;
+            }
+            $this->placeReservationsForSalesEvent->execute(
+                $this->getItemsToSell($checkoutForm, true),
+                $this->getSalesChannel(),
+                $this->getSalesEvent(self::ALLEGRO_EVENT_RESERVATION_COMPENSATED, $checkoutForm->getId())
+            );
+        } catch (\Exception $e) {
```
Comment (cieslix) [inline Model/OrderImporter/AllegroReservation.php]: why catch and throw same exception?

File: Model/Reservation.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    {
+        $this->setData(self::SKU_FIELD, $sku);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOriginalReservation(): OriginalReservationInterface
+    {
+        // TODO: Implement getOriginalReservation() method.
```
Comment (cieslix) [inline Model/Reservation.php]: ????

File: Console/Command/ImportOrder.php [hunk 5]
```diff
@@ -97,11 +97,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
         try {
             $this->processor->processOrder($checkoutForm);
         } catch (\Exception $e) {
-            $output->writeln("Error while creating/updating order with id '{$checkoutFormId}'");
+            $output->writeln("Error while creating order with id '{$checkoutFormId}'");
             $output->writeln($e);
```
Comment (cieslix) [inline Console/Command/ImportOrder.php]: writeln is expecting string, $e is an object

File: Preference/Model/ResourceModel/SaveMultiple.php [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                } else {
+                    if ($metadata['event_type'] === self::ALLEGRO_EVENT_RESERVATION_COMPENSATED) {
+                        $this->reservationRepository
+                            ->deleteBySkuAndCheckoutFormId($reservation->getSku(), $metadata['object_id']);
+                    }
+                }
+            }
+            $connection->commit();
+        } catch (\Exception $e) {
+            $connection->rollBack();
```
Comment (cieslix) [inline Preference/Model/ResourceModel/SaveMultiple.php]: please log exception

File: Model/OrderImporter/AllegroReservation.php [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+                $e
+            );
+        }
+    }
+
+    /**
+     * @param CheckoutFormInterface $checkoutForm
+     * @throws \Exception
+     */
+    public function compensateReservation(CheckoutFormInterface $checkoutForm): void
```
Comment (fsw) [inline Model/OrderImporter/AllegroReservation.php]: This method looks 100% like placeReservation witch seems like a bug or repetition, please check.

File: Model/OrderImporter/AllegroReservation.php (hunk 3 above)
Comment (fsw) [reply Model/OrderImporter/AllegroReservation.php]: I assume this is just to change error message and clean up logs and technically it is a different exception just with the same type.
I would remove this try/catch for simplicity as it seems it does not add much value but as long as $previous is passed I would not consider this a big issue.

**Pull request merged**